lib/upload.js

```diff
@@ -191,27 +199,33 @@ class Upload {
     }
 
     // Try to find the endpoint for the file in the storage
-    if (this.options.resume) {
+    if (this.options.resume && this._storage) {
       this._fingerprint = this.options.fingerprint(file, this.options);
-      let resumedUrl = Storage.getItem(this._fingerprint);
+      this._storage.getItem(this._fingerprint, (err, resumedUrl) => {
+        if (err) {
+          this._emitError(err);
+          return;
+        }
 
-      if (resumedUrl != null) {
-        this.url = resumedUrl;
-        this._resumeUpload();
-        return;
-      }
+        if (resumedUrl != null) {
+          this.url = resumedUrl;
+          this._resumeUpload();
+        } else {
+          this._createUpload();
+        }
+      });
+    } else {
+      // An upload has not started for the file yet, so we start a new one
+      this._createUpload();
     }
-
-    // An upload has not started for the file yet, so we start a new one
-    this._createUpload();
   }
 
   abort() {
     if (this._xhr !== null) {
       this._xhr.abort();
       this._source.close();
-      this._aborted = true;
```

Comment:
I figure this was a bug, since it should always set `_aborted` to true after `abort` is called

Reply:
Good catch! That would not have been a problem before this PR but might causes issues now :+1: